### PR TITLE
replace SubstitutionFormatString_TextFormat(deprecated) to SubstitutionFormatString_TextFormatSource for the accessLogFormat

### PR DESCRIPTION
**Pull request by monsteredc**
After reading the issue #34466, I found that the `SubstitutionFormatString_TextFormat` has been deprecated in Envoy v1.17.0 in favor of `SubstitutionFormatString_TextFormatSource`. This PR is to move the old syntax format to the newer API version.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 1]
```diff
@@ -233,8 +233,12 @@ func buildFileAccessLogHelper(path string, mesh *meshconfig.MeshConfig) *accessl
 		}
 		fl.AccessLogFormat = &fileaccesslog.FileAccessLog_LogFormat{
 			LogFormat: &core.SubstitutionFormatString{
-				Format: &core.SubstitutionFormatString_TextFormat{
-					TextFormat: formatString,
+				Format: &core.SubstitutionFormatString_TextFormatSource{
+					TextFormatSource: &core.DataSource{
+						Specifier: &core.DataSource_InlineString{
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:238]: What envoy version added this?

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 2]
```diff
@@ -233,8 +233,12 @@ func buildFileAccessLogHelper(path string, mesh *meshconfig.MeshConfig) *accessl
 		}
 		fl.AccessLogFormat = &fileaccesslog.FileAccessLog_LogFormat{
 			LogFormat: &core.SubstitutionFormatString{
-				Format: &core.SubstitutionFormatString_TextFormat{
-					TextFormat: formatString,
+				Format: &core.SubstitutionFormatString_TextFormatSource{
+					TextFormatSource: &core.DataSource{
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:237]: In general - using new API SGTM..

But why does the old way not work, and the new way does? I would have thought this is just a syntax change?

File: pilot/pkg/networking/core/v1alpha3/accesslog.go (hunk 1 above)
Comment (monsteredc) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go:238]: It's v1.17.0

File: pilot/pkg/networking/core/v1alpha3/accesslog.go (hunk 2 above)
Comment (monsteredc) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go:237]: Thanks. It's just a syntax change. I've investigated the code, but however cannot find why was his `accessLogFormat` field not working. I've revised the PR description, and if I can find the issue, I'll open a new PR for the fix.

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go:237]: Ah see, so this was found during https://github.com/istio/istio/issues/34466, but does not *fix* that issue, correct?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
I removed the release note as this is not a user facing change

File: pilot/pkg/networking/core/v1alpha3/accesslog.go (hunk 2 above)
Comment (monsteredc) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go:237]: correct

**Comment by monsteredc** [comment]
> I removed the release note as this is not a user facing change

@howardjohn Thanks!

**Pull request merged**